### PR TITLE
Adding Magic Leap 2 Controller Interaction Profile

### PR DESCRIPTION
**Pull request by rbessems**
This MR adds the Magic Leap 2 Controller Interaction Profile.

### CODE REVIEW

**Comment by rbessems** [comment]
@BastiaanOlij here is the Magic Leap 2 OpenXR interaction profile.

File: modules/openxr/extensions/openxr_ml2_controller_extension.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "openxr_ml2_controller_extension.h"
+#include "../action_map/openxr_interaction_profile_meta_data.h"
+
+// ML2 controllers are not part of the included OpenXR spec at the time of
```
Comment (BastiaanOlij) [inline modules/openxr/extensions/openxr_ml2_controller_extension.cpp:34]: Seeing we simply need to update the thirdparty inclusion of OpenXR headers to fix this, I suggest we do that. @akien-mga where do we stand on updating thirdparty libraries now that we're so close to RC1? If thats no problem I'll prepare a PR for moving us to OpenXR 1.0.26

**Review by BastiaanOlij — APPROVED** [review]
Looks good, just need to make a decision on whether we're going up to OpenXR 1.0.26 or if we'll move that till a later date.

File: modules/openxr/extensions/openxr_ml2_controller_extension.cpp (hunk 1 above)
Comment (akien-mga) [reply modules/openxr/extensions/openxr_ml2_controller_extension.cpp:34]: Yeah you can update the OpenXR lib, I don't expect any problem from it.

**Comment by akien-mga** [comment]
Note that the commits should be squashed. See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

But might be worth for waiting first for Bastiaan to update OpenXR.

**Comment by BastiaanOlij** [comment]
I'll update it tomorrow, today just totally got away from me, too much to do :)

File: modules/openxr/extensions/openxr_ml2_controller_extension.cpp (hunk 1 above)
Comment (BastiaanOlij) [reply modules/openxr/extensions/openxr_ml2_controller_extension.cpp:34]: Ok I'm not sure where the confusion has come from, but as far as I can see we are already including OpenXR 1.0.26 and our ML2 extension is in thirdparty:

https://github.com/godotengine/godot/blob/master/thirdparty/openxr/include/openxr/openxr.h#L4043

Comment (BastiaanOlij) [reply modules/openxr/extensions/openxr_ml2_controller_extension.cpp:34]: So @rbessems I think you can just removed the `#ifndef XR_ML_ML2_CONTROLLER_INTERACTION_EXTENSION_NAME` block?

Comment (rbessems) [reply modules/openxr/extensions/openxr_ml2_controller_extension.cpp]: @BastiaanOlij I've removed the ifdef's.

**Comment by rbessems** [comment]
@akien-mga I've squashed the MR. Ready to go now.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**